### ISSUE TITLE
pixel jitter

### ISSUE THREAD

**Issue by JanGunheD** [MiSTer-devel/Template_MiSTer#71]
Some games, especially Super Hang On, show pixel jittering.

Looking closely, all pixels jitter a tiny bit around. This becomes especially apparent when combined with filters like shadow masks.

A similar thing happened with the first split versions of the Xain'd Sleena core (just in case that might help).

Using vsync_adjust=2 (just in case this only happens in this low latency mode).

According to Jotego, this is a problem in the framework if I understand correctly:
https://github.com/jotego/jtcores/issues/226

**Comment by jotego** [comment]
I think that the problem comes from the ASCAL operating at an unrelated clock speed (is it 100MHz?) to the core pixel clock. MiSTer framework then re-samples the video signal without any proper filtering.

**Comment by paulb-nl** [comment]
The current Super Hang On core has some kind of timing issue with HDMI. The OSD is also affected which is generated after the scaler. 

All other cores work perfectly with the scaler so I don't understand why you are blaming the scaler.

HV integer scaled HDMI capture at 1080p(720p has the same issue):
![shangon_crop](https://user-images.githubusercontent.com/2227984/227732061-167bf009-fc1a-4bd9-b8dc-c66323bb805c.png)

**Comment by sorgelig** [comment]
> I think that the problem comes from the ASCAL operating at an unrelated clock speed (is it 100MHz?) to the core pixel clock. MiSTer framework then re-samples the video signal without any proper filtering.

Completely wrong assumption. Scaler doesn't resample video signal by 100MHz. 

Issues in Template should be posted ONLY if it affects many cores as some common issue. This issue doesn't happen on open source cores in MiSTer-devel.

**Comment by jotego** [comment]
> Completely wrong assumption. Scaler doesn't resample video signal by 100MHz.

But it resamples the video signal, doesn't it?

> Issues in Template should be posted ONLY if it affects many cores as some common issue. This issue doesn't happen on open source cores from MiSTer-devel.

I have seen it in cores whose pixel clock is not 6MHz. I don't know the details of the different clocks and timing considerations taken in the template. I may be doing something wrong. Maybe you are expecting the pixel clock enable to be centered at the pixel? If there are some design guidelines for cores, please kindly link to them.

**Comment by sorgelig** [comment]
> But it resamples the video signal, doesn't it?

It saves pixels into memory at core's pixel clock then read it back at HDMI clock and interpolates according to selected video filter.

> I have seen it in cores whose pixel clock is not 6MHz. I don't know the details of the different clocks and timing considerations taken in the template. I may be doing something wrong. Maybe you are expecting the pixel clock enable to be centered at the pixel? If there are some design guidelines for cores, please kindly link to them.

Issue shown on screenshots above has nothing to do with core's pixel clock or clock enable. It's happening because HDMI timings didn't meet requirements. Did you check TimeQuest Timing Analyzer Summary - is there any negative slacks? 
It may be due to design of core which may have hard to route paths so it affected HDMI part as well. Or may be some synthesizer/router settings are wrong. May be constraints are wrong. You don't use quartus IDE to compile, so it's possible some constraints are missing or MiSTer framework has modifications lead to that issue.

**Comment by birdybro** [comment]
> The current Super Hang On core has some kind of timing issue with HDMI. The OSD is also affected which is generated after the scaler.

Yes, this same exact issue occurred with outrun over HDMI for quite some time, I haven't tried the latest update from Jotego for that though.

**Comment by paulb-nl** [comment]
The current Outrun core doesn't seem to have this problem. Maybe because the core was changed from 100Mhz to 50Mhz.

**Comment by JanGunheD** [comment]
I observed the same change and suspected the added sprite framebuffer to have solved this for outrun.

**Comment by sorgelig** [comment]
Looks like fixed according to posts i've saw on FB.

**Comment by jotego** [comment]
> 
 It saves pixels into memory at core's pixel clock then read it back at HDMI clock and interpolates according to selected video filter.

This is the best approach indeed.

> Issue shown on screenshots above has nothing to do with core's pixel clock or clock enable. It's happening because HDMI timings didn't meet requirements. Did you check TimeQuest Timing Analyzer Summary - is there any negative slacks? It may be due to design of core which may have hard to route paths so it affected HDMI part as well. Or may be some synthesizer/router settings are wrong. May be constraints are wrong. You don't use quartus IDE to compile, so it's possible some constraints are missing or MiSTer framework has modifications lead to that issue.

This was very good feedback. Thank you. The timing constraints get checked and printed out if something failes in my workflow. But, I must have added a false path statement that shouldn't be there that affects the HDMI subsystem.

Thanks for the comments :-)

**Comment by JanGunheD** [comment]
It seemed to be solved for a while, now with the latest updates it is happening again.

**Comment by sorgelig** [comment]
You need to open issue in a specific core, not here. It's not an issue of template.

**Comment by jotego** [comment]
Yes, I think we you can close it here, @JanGunheD. I'm looking into it.
